Tests/LeafKitTests/GHTests/VaporLeaf.swift

```diff
@@ -50,21 +50,14 @@ final class GHLeafIssuesTest: XCTestCase {
     func testGH105() throws {
         do {
             let template = """
-            #if(1 + 1 == 2):hi#endif
+            #if((1 + 1) == 2):hi#endif
```

Comment:
This looks like we're changing the behaviour of the parser in a breaking way?

Reply:
yea, this and other minorly breaking changes are why i consider the PR leaf 5 material—some operators are deemed nonassociative and require you to add parentheses to avoid attempting to associate them.

Reply:
Why did you want to make operators non-associative, @pontaoski? Is there a technical limitation in the implementation, or was the change intentional?

Reply:
intentional: from my experience, i've seen people struggle to comprehend operators like `==` and `!=` if used in an associative way, and there's usually no reason to chain multiple of them, so making the comparison operators nonassociative forces people to add parentheses for legibility in anything more complex than `a == b` (with an error making this explicit).

but changing this would be as simple as editing the table in the scanner where the comparison operators are defined as nonassociative.

Reply:
Thanks for answering my question 🎉 

Your explanation is reasonable, and I concur that the current syntax can be confusing.

This change seems unrelated to what you set out to achieve with the rest of this PR, though? It also seems pretty contained - this might be a good candidate for a separate PR that builds on the base improvements you've made.

Given that this change would break existing production code, it'd be worthwhile socialising the idea [via the Swift forums](https://forums.swift.org/c/related-projects/vapor/30) and giving users an opportunity to provide feedback. I'm happy to write this up as a forum thread if it's not something you're keen to do.

Reply:
most of the PR is breaking changes; i honestly don't see why this one breaking change is being singled out.

Reply:
For me, it was because it was the easiest breaking change to see due to your new tests, and it was easy to see that it would affect users of Leaf (as opposed to developers working directly with LeafKit's public APIs). 

I would like to help get your changes cleaned up and merged, so if you're aware of other breaking changes that need to be handled in documentation and other consumers of this library, it would be helpful to know about them without having to go through the code line-by-line. Perhaps they could go in the PR description?

Reply:
the comparison operators are associative now